### PR TITLE
Add the name of the branch to follow

### LINKED ISSUE #1779: update_revision.sh wrongly chooses the branch to get updates from
Depending on the packages to check, scripts/utilities/update_revision.sh wrongly chooses the branch to get updates from.

For example, AT 13.0 tracks the `tbb_2019` branch of TBB, but the revision value matches more than one branch:
```
$ git branch -r --contains 4bdba61bafc6
  origin/HEAD -> origin/tbb_2020
  origin/tbb_2019
  origin/tbb_2020
```
In this case, the script chooses `HEAD` whereas `tbb_2019` is expected.

### PR DESCRIPTION
**Pull request by er-1**
Added a new variable ATSRC_PACKAGE_BRANCH to the sources file of packages
hosted in a Git repository. This avoids to guess the name of the branch from
the revision number which can lead to error from time to time.

Fix #1779

Signed-off-by: Erwan Prioul <erwan@linux.ibm.com>

### CODE REVIEW

**Comment by er-1** [comment]
The builds AT 13.0 Ubuntu18 x86_64 and AT 14.0 Ubuntu18 x86_64 failed because of a configuration issue of the worker. The worker has been updated and the builds launched again.
The builds AT 11.0 Ubuntu16 ppc64le and AT 12.0 Ubuntu16 ppc64le didn't pass a python test. Launched rebuilds to confirm.

**Comment by er-1** [comment]
> The builds AT 13.0 Ubuntu18 x86_64 and AT 14.0 Ubuntu18 x86_64 failed because of a configuration issue of the worker. The worker has been updated and the builds launched again.

The worker needs an update, but the needed package isn't available yet.

**Comment by er-1** [comment]
This will be rebased once #1872 is merged.

File: configs/12.0/packages/valgrind/sources [hunk 1]
```diff
@@ -22,6 +22,7 @@
 ATSRC_PACKAGE_NAME="Valgrind"
 ATSRC_PACKAGE_VER=3.14.0
 ATSRC_PACKAGE_REV=0c701ba2a4b1
+ATSRC_PACKAGE_BRANCH=VALGRIND_3_14_BRANCH
```
Comment (mscastanho) [inline configs/12.0/packages/valgrind/sources:25]: Is this correct? git can't seem to find this commit on that branch:
```
$ git branch -r --contains 0c701ba2a4b1
  origin/HEAD -> origin/master
  origin/VALGRIND_3_16_BRANCH
  origin/grail
  origin/master
```

**Review by mscastanho — CHANGES_REQUESTED** [review]
Thanks for working on this!
Just a small styling nit + an issue with valgrind on AT 12. I'm OK if you wait for the tests, make the change and push + merge.

I checked all commits and branches with `git branch -r --contains`, the only one I couldn't find (even though I fetched the remote branch) is the one for valgrind on AT 12. The rest is OK.

File: configs/12.0/packages/valgrind/sources (hunk 1 above)
Comment (er-1) [reply configs/12.0/packages/valgrind/sources:25]: It should be `3a3000290`, head of VALGRIND_3_14_BRANCH.

File: scripts/utilities/update_revision.sh [hunk 2]
```diff
@@ -152,29 +152,17 @@ get_latest_revision ()
...
-		local branch=$(git branch -r --contains ${ATSRC_PACKAGE_REV} \
-				   | cut -d/ -f2- \
-				   | grep -Evw "(backport|users)")
-		# When the revision is in HEAD we got a string like:
-		# master -> HEAD
-		# origin/master
-		if [[ "${branch}" = *"HEAD"* ]]; then
-			branch="HEAD"
+                local branch=${ATSRC_PACKAGE_BRANCH}
+                [[ -z "$branch" ]] && branch="HEAD"
```
[the remark this answers is not in the document]
Comment (er-1) [reply scripts/utilities/update_revision.sh]: Fixed

File: scripts/utilities/update_revision.sh [hunk 3]
```diff
@@ -152,29 +152,17 @@ get_latest_revision ()
...
+		# Since GCC moved to git (January 2020), the name of the branch
+		# has looked like: ibm/<branch>
+		if [[ $package == "gcc" ]] && [[ $configset != "next" ]]; then
+		    branch="refs/vendors/ibm/heads/${branch#*/}"
```
[the remark this answers is not in the document]
Comment (er-1) [reply scripts/utilities/update_revision.sh]: I updated the comment to make clear the IBM branches have been relocated within the vendors directory.
I also updated each `ATSRC_PACKAGE_BRANCH` from gcc `sources` files to remove the prefix `ibm/`. It eases the understanding of the script.

File: configs/12.0/packages/valgrind/sources (hunk 1 above)
Comment (mscastanho) [reply configs/12.0/packages/valgrind/sources:25]: Ah, that's unfortunate. Better not create a new problem then, we can leave it as-is.

File: scripts/utilities/update_revision.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (mscastanho) [reply scripts/utilities/update_revision.sh]: Sounds good to me. Thanks!

**Review by mscastanho — APPROVED** [review]

**Comment by mscastanho** [comment]
Same python test failure on AT 10 Ubuntu16 build, restarted it.
Not sure what happened on the AT 11 RHEL7 build. GCC stage 1 failed but all sub-steps completed successfully. I restarted it as well.

**Pull request merged**